File: frontend/src/api/advisor.ts
import { apiPost } from './client'
import type { ChatMessage } from '../stores/chatStore'
import type { InvestorProfile } from '../types/profile'
import type { ShortlistedProperty } from '../stores/shortlistStore'

export interface AdvisorResponse {
  reply: string
  red_flags: string[]
}

export function sendMessage(
  messages: ChatMessage[],
  profile: InvestorProfile,
  shortlist: ShortlistedProperty[]
): Promise<AdvisorResponse> {
  return apiPost('/api/advisor/chat', { messages, profile, shortlist })
}

File: backend/app/routers/advisor.py
from fastapi import APIRouter
from app.schemas.advisor import AdvisorRequest, AdvisorResponse
from app.services.advisor import call_advisor, detect_red_flags

router = APIRouter(prefix="/api/advisor", tags=["advisor"])


@router.post("/chat", response_model=AdvisorResponse)
def chat(req: AdvisorRequest) -> AdvisorResponse:
    latest = req.messages[-1].content
    red_flags = detect_red_flags(latest, req.profile)
    reply = call_advisor(
        [{"role": m.role, "content": m.content} for m in req.messages],
        req.profile,
        req.shortlist,
    )
    return AdvisorResponse(reply=reply, red_flags=red_flags)
